byterun/caml/mlvalues.h

```diff
@@ -134,9 +129,8 @@ bits  63        (64-P) (63-P)        10 9     8 7   0
 #define Wosize_hd(hd) ((mlsize_t) ((hd) >> 10))
 #endif /* WITH_PROFINFO */
 
-#define Hd_val(val) (((header_t *) (val)) [-1])        /* Also an l-value. */
-#define Hd_op(op) (Hd_val (op))                        /* Also an l-value. */
-#define Hd_bp(bp) (Hd_val (bp))                        /* Also an l-value. */
+#define Hd_val(val) (((header_t *) (val)) [-1] + 0)
```

Comment:
Nice trick to kill l-valueness 👍

Reply:
"Nice" might be a bit strong. It works :)

Reply:
How about something like the following, for making the de-lvaluing more explicit?

```C
#define RVALUE(x) ((void)0,x)
#define Hd_val(val) RVALUE(((header_t *) (val)) [-1])
```

Reply:
That does look like a general solution, but I'd wait until there's a second instance of this before adding another definition.